### PR TITLE
fix: Use LogicalParentOverride for PopupPanel everywhere

### PR DESCRIPTION
**Pull request by MartinZikmund**
Previously it was used only on WASM and Skia, but that caused odd issues, because routed events got propagated from Popups to the logical parents

GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 4964759</samp>

This pull request refactors the logical parent logic for UI elements by moving it from `FrameworkElement` to `DependencyObjectExtensions`. This simplifies the code and improves cross-platform consistency.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/DependencyObjectExtensions.cs
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/DependencyObjectExtensions.cs:1]: Are we breaking any test with this change? Should we add some new ones?

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/DependencyObjectExtensions.cs:1]: For now I am mainly waiting for the CI to see if it causes any trouble - if not, I will add test as needed

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/DependencyObjectExtensions.cs:1]: Should have been a draft 😁

**Review by Youssef1313 — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.4

**Pull request merged**